NuGet.Docs/Contribute/Coding-Guidelines.md

```diff
@@ -1,102 +1,511 @@
+
+### Code reviews and checkins
+
+To help ensure that only the highest quality code makes its way into the project, please submit all your code changes to GitHub as PRs. This includes Visual Studio extension, gallery and command line code changes, unit test updates, and updates to official samples. For example, sending a PR for just an update to a unit test might seem like a waste of time but the unit tests are just as important as the product code and as such, reviewing changes to them is also just as important.
+
+The advantages are numerous: improving code quality, more visibility on changes and their potential impact, avoiding duplication of effort, and creating general awareness of progress being made in various areas.
+
+In general a PR should be signed off (using the :shipit: `:shipit:` emoticon) by the Subject Matter Expert (SME) of that code. For example, a change to the Banana project should be signed off by `@MrMonkey`, and not by `@MrsGiraffe`. If you don't know the SME, please talk to one of the engineering leads and they will be happy to help you identify the SME. Of course, sometimes it's the SME who is making a change, in which case a secondary person will have to sign off on the change (e.g. `@JuniorMonkey`).
+
+To commit the PR to the repo **do not use the Big Green Button**. Instead, do a typical push that you would use with Git (e.g. local pull, rebase, merge, push).
+
+## Source code management
+
+:grey_exclamation: The *structure* of the code that we write and the *tools* that we use to write the code.
+
+### Repos
+
+To create a new repo in the https://github.com/nuget/ org, contact @yishaigalatzer.
+
+### Home repo
+
+The https://github.com/nuget/Home repo is the starting point for all things NuGet. It has the [issue tracker](https://github.com/nuget/home/issues) as well as an [up-to-date README](https://github.com/NuGet/Home/blob/master/README.md) of how repos are structured. Make sure to consult it before beginning your journey through NuGet code.
+
+### Branch strategy
+
+In general:
+
+* `master` has the code for the latest release of NuGet (e.g. alpha, beta, RC, RTM)
+* `dev` has the code that is being worked on but not yet released. This is the branch into which devs normally submit pull requests and merge changes into.
+* `release` has the code that is being staged and stabilized for an upcoming release
+
+Shortly before a release, the `release` branches are created from `dev`, and stabilization work happens there. Post-release work continus in the `dev` branch. Once the release takes place, the code is pushed from `release` to `master`.
+
+### Solution and project folder structure and naming
+
+Solution files generally go in the repo root.
+
+Solution names match repo names (e.g. NuGet.sln in the NuGet repo).
+
+Every project also needs a `project.json` and a matching `.kproj` file. This `project.json` is the source of truth for a project's dependencies and configuration options.
```

Comment:
Fixed in https://github.com/NuGet/NuGetDocs/commit/81004c2142b59f7eb84617789fdffe153d330bb5. Needs an update in https://github.com/aspnet/Home/wiki/Engineering-guidelines as well.